platform/src/main/java/org/stellar/anchor/platform/utils/TransactionHelper.java

```diff
@@ -72,6 +70,14 @@ static GetTransactionResponse toGetTransactionResponse(JdbcSep24Transaction txn)
         .amountIn(new Amount(txn.getAmountIn(), txn.getAmountInAsset()))
         .amountOut(new Amount(txn.getAmountOut(), txn.getAmountOutAsset()))
         .amountFee(new Amount(txn.getAmountFee(), txn.getAmountFeeAsset()))
+        .amountRequested(
+            new Amount(
+                txn.getRequestedAmount(),
+                "stellar:" + txn.getRequestAssetCode() + ":" + txn.getRequestAssetIssuer()))
```

Comment:
Using `stellar:` here will only work for SEP-31 transactions and SEP-24 withdrawals. Don't we want something like this? We need to support the deposit case, where the amount requested will be an off-chain asset.
```suggestion
            new Amount(txn.getRequestedAmount(), txn.getAmountRequestedAsset())
```

Reply:
It's the value passed in the request. Both withdrawal and deposit have asset_code and asset_issuer. Am I missing something? 
Maybe the logic should be
```
if (asset_issuer == null) {
 stellar:code:issuer
} else {
  code
}
```
?

Reply:
Maybe I'm misunderstanding the use case for the new attribute. I thought it was the same as Polaris' [amount_expected](https://django-polaris.readthedocs.io/en/stable/api.html#polaris.models.Transaction.amount_expected), which is always in the same units as `amount_in`.

So if its a SEP-24 deposit transaction, the asset being sent to the anchor is an offchain asset. `amount_in` is in units of the off-chain asset, and `amount_out` is in units of the Stellar asset. `amount_expected` should be the amount we expect to receive from the user.

Reply:
This is the (optional) amount that was passed to the anchor platform in the SEP request and an asset. That's it.

```
asset_code 	string 	Code of the asset the user wants to withdraw. The value passed must match one of the codes listed in the [/info](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#info) response's withdraw object


asset_code 	string 	The code of the stellar asset the user wants to receive for their deposit with the anchor. The value passed must match one of the codes listed in the [/info](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#info) response's deposit object.
```

Reply:
I think I get what you are saying. Should we just return asset_code and stellar:code:issuer if issuer is non-null?

Reply:
Ok, so then this field is different than `amount_expected`. `amount_requested` will be the amount received by the user in the deposit case, and the amount received by the anchor in the withdraw case. Lets just make sure thats documented properly.

Reply:
I'm just confused on why we're adding this field. Can you provide some context on why we need it?

Reply:
I think we can just use amount_in and asset_in instead. I don't remember the details on why I wanted to add it

Reply:
The `amountRequested` has different asset representation with `amountIn` and `amountOut`. We should have the same representation.

Reply:
This is fixed in the latest commit. Now assets follow the logic:
1. If is set in db, return db data
2. If not, use `assetService` to query (using request asset code and issuer)
For `amountExpected` field, we always use request data.